framework/mesh/mesh_continuum/mesh_continuum.cc

```diff
@@ -652,4 +653,88 @@ MeshContinuum::SetBoundaryIDFromLogical(const LogicalVolume& log_vol,
     grid_bndry_id_map[bndry_id] = boundary_name;
 }
 
+void
+MeshContinuum::ComputeVolumePerMaterialID() const
+{
+  auto grid_ptr = GetCurrentMesh();
+  const auto& grid = *grid_ptr;
+  std::shared_ptr<SpatialDiscretization> sdm_ptr = PieceWiseLinearContinuous::New(grid);
+  const auto& sdm = *sdm_ptr;
+
+  // Create a map to hold local volume with local material as key
+  std::map<int, double> material_volumes;
+  for (auto& cell : local_cells)
+  {
+    const auto& cell_mapping = sdm.GetCellMapping(cell);
+    const double volume = cell_mapping.CellVolume();
+    material_volumes[cell.material_id] += volume;
+  }
+
+  // Collect all local material IDs
+  std::set<int> unique_material_ids;
+  for (const auto& pair : material_volumes)
+    unique_material_ids.insert(pair.first);
+  // convert set to vector
+  std::vector<int> local_material_ids(unique_material_ids.begin(), unique_material_ids.end());
+  int local_size = local_material_ids.size();
+
+  // Initialize vector to hold sizes from all processes
+  std::vector<int> all_sizes(mpi_comm.size());
+  // Gather all local material ID sizes from all processes
+  mpi_comm.all_gather(local_size, all_sizes);
+
+  // Compute the displacement and total size
+  std::vector<int> displs(mpi_comm.size(), 0); // Initialize displacement vector
+  int total_size = 0;
+  for (int i = 0; i < mpi_comm.size(); ++i)
+  {
+    displs[i] = total_size;
+    total_size += all_sizes[i];
+  }
+
+  // Initialize vector to hold all material IDs from all processes
+  std::vector<int> all_material_ids(total_size);
```

Comment:
```suggestion
  std::vector<int> global_material_ids(total_size);
```

Reply:
the reoson things look a bit convoluted in that routine is because a sub-domain may only have a few of the material IDs. So the `local_material_ids` are the ones we found in that partition, while  `all_material_ids` is the union of all the `local_material_ids`. The ids themselves do not changed values (contrarily to local_dof and global_dof), so i didn't want to give the local-vs-global flavor to the names.

Reply:
I have removed  `all_material_ids` and re-assigned it. should be good now